### PR TITLE
Refactoring resource not found throws and single query to fetch all posts

### LINKED ISSUE #42: Procurar solução mais elegante para os sorts

### PR DESCRIPTION
**Pull request by CarlosFeldmann**
feat: BaseRepository class for helper methods
feat: database initialization with data.sql
feat: query to fetch all posts from sellers that a given customer follows
fix: removing useless relationship in product to post

### CODE REVIEW

File: src/main/resources/data.sql [hunk 1]
```diff
@@ -0,0 +1,25 @@
+INSERT INTO PUBLIC.USER (DTYPE, USER_ID, USER_NAME) VALUES ('Customer', 1, 'Ednaldo');
+INSERT INTO PUBLIC.USER (DTYPE, USER_ID, USER_NAME) VALUES ('Customer', 2, 'Carlos');
+INSERT INTO PUBLIC.USER (DTYPE, USER_ID, USER_NAME) VALUES ('Customer', 3, 'Luiz');
+INSERT INTO PUBLIC.USER (DTYPE, USER_ID, USER_NAME) VALUES ('Customer', 4, 'Giovana');
```
Comment (Giovannapls) [inline src/main/resources/data.sql]: Acho que o nome Giovanna com dois N é melhor

**Review by Victor-Planas — APPROVED** [review]
Nice 👌

File: src/main/resources/data.sql [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+INSERT INTO PUBLIC.FOLLOW_DATE (USER_FOLLOWED_ID, USER_FOLLOWER_ID, DATE) VALUES (6, 4, '2021-07-08');
+
+
+INSERT INTO PUBLIC.PRODUCT (PRODUCT_ID, BRAND, COLOR, NOTES, PRODUT_NAME, TYPE) VALUES (1, 'Logitech', 'Preto', 'É um mouse', 'Mouse Gamer RGB PVP', 'Gamer');
+INSERT INTO PUBLIC.PRODUCT (PRODUCT_ID, BRAND, COLOR, NOTES, PRODUT_NAME, TYPE) VALUES (2, 'Razer', 'Azul', 'DAHORA!', 'Teclado RGB', 'GAMER');
+INSERT INTO PUBLIC.PRODUCT (PRODUCT_ID, BRAND, COLOR, NOTES, PRODUT_NAME, TYPE) VALUES (3, 'Apple', 'Prata', 'Macbook Pro 2019', 'Macbook Pro 2019', 'Seilaneh');
+INSERT INTO PUBLIC.PRODUCT (PRODUCT_ID, BRAND, COLOR, NOTES, PRODUT_NAME, TYPE) VALUES (4, 'Logitech', 'Preto', 'Bom pra jogar euro truck', 'Volante G920', 'Gamer');
+
+INSERT INTO PUBLIC.POST (DTYPE, POST_ID, CATEGORY, DATE, PRICE, DISCOUNT, HAS_PROMO, PRODUCT_ID, SELLER_USER_ID) VALUES ('PromotionPost', 1, 21, '2021-07-03', 250.0, 0.12, true, 1, 5);
+INSERT INTO PUBLIC.POST (DTYPE, POST_ID, CATEGORY, DATE, PRICE, DISCOUNT, HAS_PROMO, PRODUCT_ID, SELLER_USER_ID) VALUES ('PromotionPost', 2, 31, '2021-07-02', 1200.0, 0.15, true, 2, 6);
```
Comment (Giovannapls) [inline src/main/resources/data.sql:21]: Teclado muito caro, não compraria

**Review by Giovannapls — APPROVED** [review]
🤌🤌

**Review by lmarizmeli — APPROVED** [review]

**Pull request merged**